### ISSUE TITLE
measure-perf/smt: segfault on perf. measurement via mediasdk samples on HEVC encoding

### ISSUE THREAD

**Issue by dvrogozh** [intel/media-delivery#43]
```
$ docker build \
  $(env | grep -E '(_proxy=|_PROXY)' | sed 's/^/--build-arg /') \
  --file Dockerfile.ubuntu \
  --tag intel-media-delivery \
  .
$ docker run  -it --rm --device=/dev/dri/renderD128 --group-add=39 --cap-add SYS_ADMIN intel-media-delivery .

$ ffmpeg -an -hwaccel qsv -qsv_device /dev/dri/renderD128 \
  -c:v h264_qsv -i /opt/data/embedded/WAR_TRAILER_HiQ_10_withAudio.mp4 \
  -vf scale_qsv=w=-1:h=2160,vpp_qsv=framerate=60 \
  -c:v hevc_qsv -preset medium -profile:v main -b:v 1000000 -vframes 20 \
  -y /tmp/WAR.hevc

$ measure perf --verbose --skip-perf /tmp/WAR.hevc
#####################################################################
MSPERF (MULTI STREAMS PERFORMANCE) v0.20.06.03
#####################################################################

ACCEPTED !!! to continue with [utilization/metric/statistic] profiles


#####################################################################
 Profiling: WAR.hevc
 PASS: via FFMPEG/FFPROBE
 content_fps = 60.0 , content_height = 2160 , content_codec = hevc



#####################################################################
PNP MEDIA SMT HEVC-AVC: MULTISTREAM: 1 & ITERATION: 0 - WAR
#####################################################################
 [VERBOSE][CMD] /usr/share/mfx/samples/sample_multi_transcode -fps 60 -i::h265 /tmp/WAR.hevc -hw -async 1 -u 4 -gop_size 256 -dist 8 -num_ref 5 -vbr -b 9000 -NalHrdConformance:off -VuiNalHrdParameters:off -hrd 5000 -InitialDelayInKB 2500 -extbrc::implicit -ExtBrcAdaptiveLTR:on -o::h264 /opt/data/artifacts/measure/perf/output/WAR_hevc_0.h264  -device /dev/dri/renderD128 -p /tmp/perf/WAR_hevc_0_transcode_log.txt >> /tmp/perf/console_log.txt 2>&1

 [VERBOSE][LINUX_PERF_TOOLS] /opt/intel/samples/bin/performance/MSGo.py

 Exit early, hang/error in submitted commands: 239
```

NOTE: ~~AVC looks working fine.~~ < this was wrong, I mixed with different input stream, this seems input dependent

**Comment by dvrogozh** [comment]
This is related to extbrc:
```
# WORKS:
/usr/share/mfx/samples/sample_multi_transcode -fps 60 -i::h265 /tmp/WAR.hevc \
  -hw -async 1 -u 4 -gop_size 256 -dist 8 -num_ref 5 -vbr -b 9000 \
  -NalHrdConformance:off -VuiNalHrdParameters:off -hrd 5000 -InitialDelayInKB 2500 \
  -extbrc::implicit -ExtBrcAdaptiveLTR:on -o::h264 /opt/data/artifacts/measure/perf/output/WAR_hevc_0.h264  \
  -device /dev/dri/renderD128

# FAILS (no -extbrc::implicit):
/usr/share/mfx/samples/sample_multi_transcode -fps 60 -i::h265 /tmp/WAR.hevc \
  -hw -async 1 -u 4 -gop_size 256 -dist 8 -num_ref 5 -vbr -b 9000 \
  -NalHrdConformance:off -VuiNalHrdParameters:off -hrd 5000 -InitialDelayInKB 2500 \
   -ExtBrcAdaptiveLTR:on -o::h264 /opt/data/artifacts/measure/perf/output/WAR_hevc_0.h264 \
   -device /dev/dri/renderD128
```

**Comment by dvrogozh** [comment]
This is regression between 20.2.1 and 20.3.0 msdk releases.

**Comment by dvrogozh** [comment]
@jamesiwan : fyi

**Comment by dvrogozh** [comment]
We fail with -54 error (CM_GPUCOPY_INVALID_SURFACES) from CM. It encounters in couple of places:
```
media_driver/agnostic/common/cm/cm_queue_rt.cpp:1396:        return CM_GPUCOPY_INVALID_SURFACES;
media_driver/agnostic/common/cm/cm_queue_rt.cpp:2008:        return CM_GPUCOPY_INVALID_SURFACES;
media_driver/agnostic/common/cm/cm_queue_rt.cpp:2018:            return CM_GPUCOPY_INVALID_SURFACES;
```

So, setting gdb breakpoints we encounter this:
```
Breakpoint 2, CMRT_UMD::CmQueueRT::EnqueueCopyGPUToGPU (this=0x555555d9e010, outputSurface=0x555555e35058, inputSurface=0x7fffdc000d28, option=0, event=@0x7fffee6df8f8: 0x0)
    at /home/dvrogozh/git/github/media-driver/media_driver/agnostic/common/cm/cm_queue_rt.cpp:2008
2008            return CM_GPUCOPY_INVALID_SURFACES;
(gdb) p dstSurfaceWidth
$1 = 3840
(gdb) p srcSurfaceWidth
$2 = 3840
(gdb) p dstSurfaceHeight
$3 = 2160
(gdb) p srcSurfaceHeight
$4 = 2176
(gdb) p dstSizePerPixel
$5 = 1
(gdb) p srcSizePerPixel
$6 = 1
```

I.e. we fail on this condition: https://github.com/intel/media-driver/blob/2c12eac7bdd5bb5ab97c740c9ad66cfaa9ba23be/media_driver/agnostic/common/cm/cm_queue_rt.cpp#L2008"
```
    if ((dstSurfaceWidth != srcSurfaceWidth) ||
        (dstSurfaceHeight < srcSurfaceHeight) ||  //relax the restriction
        (dstSizePerPixel != srcSizePerPixel))
    {
        CM_ASSERTMESSAGE("Error: Size of dest surface does not match src surface.");
        return CM_GPUCOPY_INVALID_SURFACES;
    }
```

So, the root cause is simple: MCTF does not account for surface alignment.

**Comment by dvrogozh** [comment]
This got fixed w/ 20.3.1 msdk release in a form of switching off MCTF. And updated package is available in the repositories.intel.com. For the sake of media-delivery this is enough to close the issue. We are awaiting for the msdk release update with re-enabled MCTF however.
